packages/block-editor/src/components/block-toolbar/style.scss

```diff
@@ -50,13 +58,19 @@
 		}
 	}
 
-	&.is-synced,
-	&.is-connected {
-		.block-editor-block-switcher .components-button .block-editor-block-icon {
-			color: var(--wp-block-synced-color);
+	// Disabled block toolbar icon should have a fixed color.
+	.components-button.block-editor-block-toolbar__block-icon[aria-disabled="true"] {
+		color: $gray-900;
+
+		// Since it's not clickable, though, don't show a hover state.
+		&:hover {
+			color: $gray-900;
```

Comment:
Moved from the block switcher as it no longer handles the button when there is no switcher.